crates/bevy_window/src/windows.rs

```diff
@@ -1,36 +1,69 @@
 use super::{Window, WindowId};
 use bevy_utils::HashMap;
 
+/// A collection of [`Window`]s with unique [`WindowId`]s.
 #[derive(Debug, Default)]
 pub struct Windows {
     windows: HashMap<WindowId, Window>,
 }
 
 impl Windows {
+    /// Add the provided [`Window`] to the [`Windows`] resource.
     pub fn add(&mut self, window: Window) {
         self.windows.insert(window.id(), window);
     }
 
+    /// Get a reference to the provided [`WindowId`].
     pub fn get(&self, id: WindowId) -> Option<&Window> {
         self.windows.get(&id)
     }
 
+    /// Get a mutable reference to the provided [`WindowId`].
     pub fn get_mut(&mut self, id: WindowId) -> Option<&mut Window> {
         self.windows.get_mut(&id)
     }
 
+    /// Get a reference to the primary [`Window`].
     pub fn get_primary(&self) -> Option<&Window> {
         self.get(WindowId::primary())
     }
 
+    /// Get a reference to the primary [`Window`].
+    ///
+    /// # Panics
+    ///
+    /// Panics if the primary window does not exist in [`Windows`]
+    pub fn primary(&self) -> &Window {
+        self.get_primary().expect("Primary window does not exist")
+    }
+
+    /// Get a mutable reference to the primary [`Window`].
     pub fn get_primary_mut(&mut self) -> Option<&mut Window> {
         self.get_mut(WindowId::primary())
     }
 
+    /// Get a mutable reference to the primary [`Window`].
+    ///
+    /// # Panics
+    ///
+    /// Panics if the primary window does not exist in [`Windows`]
+    pub fn primary_mut(&mut self) -> &mut Window {
+        self.get_primary_mut()
+            .expect("Primary window does not exist")
+    }
+
+    /// Returns the scale factor of the primary window, or `1.0` if the window does not exist.
+    pub fn primary_scale_factor(&self) -> f64 {
+        self.get_primary()
+            .map_or_else(|| 1.0, |window| window.scale_factor())
```

Comment:
```suggestion
            .map_or(1.0, |window| window.scale_factor())
```

Reply:
Isn't `or_else` preferred because it isn't eagerly evaluated?